### PR TITLE
refactor: upgrade to junit5

### PR DESCRIPTION
**Pull request by qigao**
- [x] upgrade from junit4 to junit5
- [ ] ~~mv annotation to events package~~
- [ ] ~~mv interface ,abstract class and enum to api package~~
- [x] fix java.version in pom.xml

Fixes: #31

### CODE REVIEW

**Comment by xXNurioXx** [comment]
Thanks for your work, I will ask you to reorganize your commits a little bit. Also, this PR is doing too many things. In order to avoid getting rejected, you must try to only do only change at the time.

For example, if you change the project packages, it will break backward compatibility. Making all applications using this library unable to upgrade easily. That could be something great, but in this case, is not worth it.

That's the kind of unnecessary change that is blocking your PR from getting merged.

If you found that there is some space for improvements, you can create multiple PRs and that will avoid one 'not accepted' change blocking the rest. 

Also, you introduced some spaces in front of classes and methods declarations and blank lines that don't make sense. You must give important attention to that kind of detail.

And last but not less important. Embrace having many commits, one for each kind of change you apply. Even if you add too many commits, squashing them is by far more easy than decoupling a big commit.

File: src/test/java/me/nurio/events/eventdispathevent/EventDispatchEventCancellationTest.java [hunk 1]
```diff
@@ -1,42 +1,43 @@
...
-
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 
 /**
- * This test will ensure that the EventDispatchEvent cancellation prevents the event from being dispatched.
+ * This test will ensure that the EventDispatchEvent cancellation prevents the event from being
+ * dispatched.
```
Comment (xXNurioXx) [inline src/test/java/me/nurio/events/eventdispathevent/EventDispatchEventCancellationTest.java:20]: Try to avoid editing lines that you don't need to update. The less you change, fewer things can cause chances of not getting merged.

This repeats all over the tests.

File: src/test/java/me/nurio/events/internal/annotations/EventAnnotationHandlersManagerReflectionTest.java [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
+import org.junit.jupiter.api.Test;
 
 /**
  * This class will test the EventRefection behavior with valid and invalid event handlers.
  */
-public class EventAnnotationHandlersManagerReflectionTest {
+ class EventAnnotationHandlersManagerReflectionTest {
 
-    private EventAnnotationHandlersManager handlersManager;
+    private static EventAnnotationHandlersManager handlersManager;
```
Comment (xXNurioXx) [inline src/test/java/me/nurio/events/internal/annotations/EventAnnotationHandlersManagerReflectionTest.java:23]: Why does this need to be static? Try to avoid unnecessary changes. :)

File: src/test/java/me/nurio/events/PerformanceTest.java [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
 /**
  * This class is used to manually perform fast benchmarking of the call event process.
  */
-@Ignore
-public class PerformanceTest {
+@Disabled
+ class PerformanceTest {
 
-    private EventManager eventManager = new ReflectedEventManager();
+    private final EventManager eventManager = new ReflectedEventManager();
```
Comment (xXNurioXx) [inline src/test/java/me/nurio/events/PerformanceTest.java:17]: This is a disabled test, it's something internal. Also, there is a big debate about if using finals or not, so not always is a good idea to make it final.

If your main change doesn't need to change other things, why touch it? :)

**Review by xXNurioXx — CHANGES_REQUESTED** [review]
I added some comments and examples of possible problems with your PR. If you fix all of them, I will merge the Junit5 upgrade. 😄 ❤️ 

Please, undo all the changes related to packages, etc.

**Comment by qigao** [comment]
Hi @xXNurioXx ,
 just pushed my update as you advised, thanks for the patience  and good advice.

**Review by xXNurioXx — APPROVED** [review]

**Comment by xXNurioXx** [comment]
Thanks for your contribution! Merging it!
